lib/auth/token_provider.dart

```diff
@@ -53,7 +54,10 @@ class TokenProvider {
 
   Future _refresh() async {
     var response = await client.post(
-      Uri.parse('https://securetoken.googleapis.com/v1/token'),
+      useEmulator
+          ? Uri.parse(
+              'http://localhost:9099/securetoken.googleapis.com/v1/token')
```

Comment:
Might be worth noting in a comment somewhere that the auth emulator is hardcoded for localhost and port 9099 (this is configurable via the firebase emulator suite, some users might have something different). I personally didn't plumb a port configuration down because it seemed like overkill, but it could be useful. Usually the config takes some 'Host,Port' object that it passes down (like firedart already does in firestore).

Reply:
Good call, I will do that, thanks!